faunadb/stream_test.go

```diff
@@ -0,0 +1,303 @@
+package faunadb_test
+
+import (
+	"io"
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/suite"
+
+	f "github.com/fauna/faunadb-go/v3/faunadb"
+)
+
+func TestRunStreamTests(t *testing.T) {
+	suite.Run(t, new(StreamsTestSuite))
+}
+
+type StreamsTestSuite struct {
+	suite.Suite
+	client *f.FaunaClient
+}
+
+var (
+	streamCollection f.RefV
+)
+
+type counterMutex struct {
+	mu sync.Mutex
+	i  int64
+}
+
+func (c *counterMutex) Inc() {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	c.i = c.i + 1
+}
+
+func (c *counterMutex) Value() int64 {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	return c.i
+}
+
+func (s *StreamsTestSuite) TestStreamDocumentRef() {
+	//var wg sync.WaitGroup
+	var subscription f.StreamSubscription
+
+	ref := s.createDocument()
+
+	subscription = s.client.Stream(ref)
+	subscription.Start()
+	for evt := range subscription.EventsMessages() {
```

Comment:
```suggestion
	for evt := range subscription.Events() {
```

I think it looks better if it's just called `Events`?

Reply:
We've already have Events in couple of other places. W/hat do you think about StreamEvents ?